SIMD_and_vectorization.md

```diff
@@ -124,6 +124,38 @@ A developer wanting to build an application or library that can detect the suppo
 
 ## Porting codes with SSE/AVX intrinsics to NEON
 
+### Detecting arm64 systems
+
+Projects may fail to build on arm64 with `error: unrecognized command-line
+option '-msse2'`, or `-mavx`, `-mssse3`, etc.  These compiler flags enable x86
+vector instructions.  Compilers for arm64 do not recognize these flags.  The
+presence of this error means that the configure scripts have not detected arm64
+as the target system.
+
+There are two types of build systems: native compilation and cross-compilation.
```

Comment:
this i making an assumption that everyone is using autotools/configure. I doubt these are the cases where we've seen ppl have problems, it's mostly makefiles w/o any cflags guards or CMake without any tests/platform checks

Reply:
My intention was to describe the configure systems independent of whether it's autotools or cmake.  What I wrote should apply to both systems.  Writing something more specific for cmake or autotools is difficult as each project has their own way to detect platforms.

Reply:
I simplified the text of this section.
I added the code to detect arm64 systems.